### ISSUE TITLE
Provide TypeScript definitions

### ISSUE THREAD

**Issue by legrego** [w33ble/rison-node#1]
Adding TS definitions would help with type safety

**Comment by fabio-looker** [comment]
Is this what would need to be added? https://www.npmjs.com/package/@types/rison
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/rison

**Comment by legrego** [comment]
Yep, I think those typedefs would end up working for this. We'd just need to re-publish under `@types/rison-node`, or include those typedefs directly in this repository. I prefer the latter approach, as it's one less dependency to maintain in downstream projects

**Comment by w33ble** [comment]
I'm also in favor of pulling types into this repo, less dependencies and also easier to keep things in sync.

**Comment by tiffon** [comment]
Btw `decode_uri` is not in the linked typedefs since it is specific to rison-node.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/761c11c125ae9acf6d9450959f13c01bb1c3e6f1/types/rison/index.d.ts#L32-L36
